lib/fluent/plugin/in_tail.rb

```diff
@@ -1059,7 +1055,27 @@ def read_lines(lines)
             rbuf = @buffer.slice(0, idx + 1)
             @buffer = @buffer.slice(idx + 1, @buffer.size)
             idx = @buffer.index(@eol)
-            lines << convert(rbuf)
+
+            is_long_line = !@max_line_size.nil? && (
+              rbuf.bytesize > @max_line_size || @was_long_line
+            )
+
+            if !is_long_line
+              lines << convert(rbuf)
+            else
+              @log.warn "received line length is longer than #{@max_line_size}"
+              @log.debug "skipped line"
+              @was_long_line = false
+            end
+          end
+
+          is_long_line = !@max_line_size.nil? && (
+            @buffer.bytesize > @max_line_size || @was_long_line
+          )
+
+          if is_long_line
+            @buffer = ''.force_encoding(@from_encoding)
```

Comment:
```suggestion
            @buffer.clear
```

Reply:
Hi @daipom

I discovered that the change in this suggestion(https://github.com/fluent/fluentd/pull/4491#discussion_r1615863064) causes the following test to fail:

```
  data(
    small: ["128", 8192],
  )
  test 'max_line_size_8192' do |(label, size)|
    config = config_element("", "", {
                              "tag" => "max_line_size",
                              "path" => "#{@tmp_dir}/with_long_lines.txt",
                              "format" => "none",
                              "read_from_head" => true,
                              "max_line_size" => label,
                              "log_level" => "debug"
                            })
    Fluent::FileWrapper.open("#{@tmp_dir}/with_long_lines.txt", "w+") do |f|
      f.puts "foo"
      f.puts "x" * size # 'x' * size + \n > @max_line_size
      f.puts "bar"
    end
    d = create_driver(config, false)
    timestamp = Time.parse("Mon Nov 29 11:22:33 UTC 2021")
    Timecop.freeze(timestamp)
    d.run(expect_records: 2)
    assert_equal([
                    [{"message" => "foo"},{"message" => "bar"}],
                    [
                      "2021-11-29 11:22:33 +0000 [warn]: received line length is longer than #{label}\n",
                      "2021-11-29 11:22:33 +0000 [debug]: skipped line: #{'x' * size}\n"
                    ]
                  ],
                  [
                    d.events.collect { |event| event.last },
                    d.logs[-3..]
                  ])
  end
```

When the line_size is greater than max_line_size and line_size is greater than or equal to 8192, the test fails.

To log the long line even if it is skipped, you can log the line above this code like this:

```
          if is_long_line
            @log.warn "received line length is longer than #{@max_line_size}"
            @log.debug("skipped line: ") { convert(@buffer).chomp }
            @buffer.clear
            @was_long_line = true
          end
```

If you have any idea, please let me know.

Thank you for your review!!

Reply:
I see! Thanks for checking this behavior!

`@log.debug("skipped line: ") { convert(rbuf).chomp }` sometimes can not output the entire line that is expected.
It is because some parts of the line are already cleared if `@was_long_line` is `true`, 

I think this is simply a debug log issue, right?
It is less important, so it would be good if we could fix it within the possible range.